### DISCUSSION TITLE
Preparing subpixel scaled images.

### DISCUSSION THREAD

**Discussion by mcbp223** [Bodmer/TFT_eSPI#2215] [General]
To go along with the subpixel drawing primitives in the library, we can use background images scaled down to subpixel level, with the help of [ImageWorsener](https://github.com/jsummers/imageworsener), by [Jason Summers](http://entropymine.com/imageworsener/subpixel/).

Here are some images taken from the internet (copyrights may apply), scaled down to 240x240.

For comparison, downscaled with and without subpixel rendering, both using Lanczos filter.
`imagew -S 240,240 -depth 5,6,5 -offsetrb 1/3 -filter lanczos <source> <dest>`

![RE-05b-lanc](https://user-images.githubusercontent.com/24973142/204461430-98dfaef1-011b-45ea-b48c-55f4e9703a58.png)

![RE-05b-smrs-lanc](https://user-images.githubusercontent.com/24973142/204461500-182f6470-de3a-43e8-bfc9-742800009084.png)

Test pattern images:
![test-pattern__1340_c_smrs-lanc_sq flip rgb565 swap](https://user-images.githubusercontent.com/24973142/204461829-a5a18ae1-9feb-4385-83d0-1910a891f9f6.png)

Digital fulldome test pattern by [Paul Bourke](http://paulbourke.net/dome/testpattern/).
![fulldome_test-pattern_smrs-lanc_240 flip rgb565 swap](https://user-images.githubusercontent.com/24973142/204461932-ceea5c8a-26a5-47cc-b7dd-0477b8aef74e.png)

----------
I've uploaded a Python [script here](https://gist.github.com/mcbp223/89ca2899e6f314a9cc4240ecd96bc51e) to help with conversion into C-style arrays, rgb565 byte swapping and vertical flip.

**Comment by Bodmer** [comment]
Thanks for that.
